hircluster.c

```diff
@@ -960,8 +960,8 @@ dict *parse_cluster_nodes(redisClusterContext *cc, char *str, int str_len,
                 goto error;
             }
 
-            // the address string is ":0", skip this node.
-            if (sdslen(part[1]) == 2 && strcmp(part[1], ":0") == 0) {
+            // if the address string starts with ":0", skip this node.
+            if (sdslen(part[1]) >= 2 && part[1][0] == ':' && part[1][1] == '0') {
```

Comment:
Clang-format complains that the line is too long. But we could also make it shorter and more readable with memcmp:

```suggestion
            if (sdslen(part[1]) >= 2 && memcmp(part[1], ":0", 2) == 0) {
```